### PR TITLE
Dockerfile: Configure ScanCode to use Python 3 even if Python 2 is installed

### PR DESCRIPTION
**Pull request by sschuberth**
Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

**Comment by sschuberth** [comment]
@pombredanne is this the right / best way to force ScanCode to use Python 3 even if Python 2 is installed?

**Comment by pombredanne** [comment]
> @pombredanne is this the right / best way to force ScanCode to use Python 3 even if Python 2 is installed?

yes, that's the way.

Note also that you should also create the initial license index in your Dockerfile IMHO
For this run scancode once with this:
 `scancode   --reindex-licenses`
This will check the license index cache and reindex if needed and exit.

File: Dockerfile [hunk 1]
```diff
@@ -145,7 +145,7 @@ RUN /opt/ort/bin/import_proxy_certs.sh && \
     curl -ksSL https://github.com/nexB/scancode-toolkit/archive/v$SCANCODE_VERSION.tar.gz | \
         tar -zxC /usr/local && \
         # Trigger configuration for end-users.
-        /usr/local/scancode-toolkit-$SCANCODE_VERSION/scancode --version && \
+        PYTHON_EXE=/usr/bin/python3 /usr/local/scancode-toolkit-$SCANCODE_VERSION/scancode --version && \
```
Comment (fviernau) [inline Dockerfile:148]: Would it make sense to leave a code comment saying that this actually configures ScanCode to use Python 3 ?

Comment (sschuberth) [reply Dockerfile:148]: I'd say that's pretty much obvious, due to a lack of alternatives what else it could do, or?

Comment (sschuberth) [reply Dockerfile:148]: But I'll amend the existing comment anyway.

**Review by fviernau — APPROVED** [review]

**Comment by sschuberth** [comment]
BTW @pombredanne, are there any significant benefits of running ScanCode 3.2 with Python 3 compared to Python 2?

**Comment by pombredanne** [comment]
> BTW @pombredanne, are there any significant benefits of running ScanCode 3.2 with Python 3 compared to Python 2?

Nothing special: It is slightly but marginally faster, but the biggest benefit is that Python 2 is legacy and ScanCode support for Python 2 will be dropped as soon as 3.2 is final and only Python 3 will be supported going forward.
So rather than a benefit, this will become a mandate anytime :)

**Comment by pombredanne** [comment]
Side note: you should experience major startup performance boost from having `scancode --reindex-licenses` in the Dockerfile and the index saved in the container image.

**Pull request merged**